### DISCUSSION TITLE
Improve performance of function `rationalize`

### DISCUSSION THREAD

**Discussion by Divyesh-Shah-17** [josdejong/mathjs#3261] [Ideas]
**Describe the bug**
The rationalize function is experiencing significant performance issues, causing noticeable slowness. In some cases, this slowness leads to the browser crashing.

**To Reproduce**
I encountered this issue with a different example, but I found a similar problem in the documentation with the following equation:
`math.rationalize('x/(1-x)/(x-2)/(x-3)/(x-4) + 2x/ ( (1-2x)/(2-3x) )/ ((3-4x)/(4-5x) )')`

**Comment by josdejong** [comment]
Thanks for reporting. There may be a clear culprit or some infitite loop occurring there.

Help debugging this issue would be very welcome.

**Comment by josdejong** [comment]
I've done some debugging, but I do not see an obvious issue causing this performance issue. I think the logic works fine but is just to inefficient to be able to process an expression this complex, making it take forever. I'll move this issue into a discussion about improving the performance.

For reference: there is already a skipped unit test with this example:

https://github.com/josdejong/mathjs/blob/d1ecf44e2f4dba4966d602851329643eb8a788c7/test/unit-tests/function/algebra/rationalize.test.js#L126-L131
